lib/icinga/clusterevents.cpp

```diff
@@ -619,6 +619,68 @@ Value ClusterEvents::AcknowledgementClearedAPIHandler(const MessageOrigin::Ptr&
 
 Value ClusterEvents::ExecuteCommandAPIHandler(const MessageOrigin::Ptr& origin, const Dictionary::Ptr& params)
 {
+	ApiListener::Ptr listener = ApiListener::GetInstance();
+	if (!listener)
+		return Empty;
+
+	if (params->Contains("endpoint")) {
+		Endpoint::Ptr execEndpoint = Endpoint::GetByName(params->Get("endpoint"));
+		if (execEndpoint != Endpoint::GetLocalEndpoint()) {
```

Comment:
What if the first condition is true and the second one false?

Reply:
The execution will be enqueued, is it wrong?

Reply:
No, it's OK. 🤦‍♂️